api/v1/client.go

```diff
@@ -632,6 +639,128 @@ func (c *storageClient) RuntimeBlocks(ctx context.Context, r *http.Request) (*st
+	var q storage.RuntimeTransactionsRequest
+	params := r.URL.Query()
+	if v := params.Get("block"); v != "" {
+		block, err := validateInt64(v)
+		if err != nil {
+			return nil, common.ErrBadRequest
+		}
+		q.Block = &block
+	}
+
+	p, err := common.NewPagination(r)
+	if err != nil {
+		c.logger.Info("pagination failed",
+			"request_id", ctx.Value(storage.RequestIDContextKey),
+			"err", err.Error(),
+		)
+		return nil, common.ErrBadRequest
+	}
+
+	sts, err := c.storage.RuntimeTransactions(ctx, &q, &p)
+	if err != nil {
+		return nil, err
+	}
+
+	var ts RuntimeTransactionList
+	for _, st := range sts.Transactions {
+		var utx types.UnverifiedTransaction
+		if err = cbor.Unmarshal(st.Raw, &utx); err != nil {
+			return nil, fmt.Errorf("round %d tx %d utx unmarshal: %w", st.Round, st.Index, err)
+		}
+		tx, err := uncategorized.OpenUtxNoVerify(&utx)
+		if err != nil {
+			return nil, fmt.Errorf("round %d tx %d utx open no verify: %w", st.Round, st.Index, err)
+		}
+		sender0, err := uncategorized.StringifyAddressSpec(&tx.AuthInfo.SignerInfo[0].AddressSpec)
+		if err != nil {
+			return nil, fmt.Errorf("round %d tx %d signer 0: %w", st.Round, st.Index, err)
+		}
+		var cr types.CallResult
+		if err = cbor.Unmarshal(st.ResultRaw, &cr); err != nil {
+			return nil, fmt.Errorf("round %d tx %d result unmarshal: %w", st.Round, st.Index, err)
+		}
+		t := RuntimeTransaction{
+			Round:   st.Round,
+			Index:   st.Index,
+			Hash:    st.Hash,
+			EthHash: st.EthHash,
+			// TODO: Get timestamp from that round's block
+			Sender0:   sender0,
+			Nonce0:    tx.AuthInfo.SignerInfo[0].Nonce,
+			FeeAmount: tx.AuthInfo.Fee.Amount.Amount.String(),
+			FeeGas:    tx.AuthInfo.Fee.Gas,
+			Method:    tx.Call.Method,
+			Body:      tx.Call.Body,
+			Success:   cr.IsSuccess(),
+		}
+		if err = uncategorized.VisitCall(&tx.Call, &cr, &uncategorized.CallHandler{
+			AccountsTransfer: func(body *accounts.Transfer) error {
+				to, err2 := uncategorized.StringifySdkAddress(&body.To)
+				if err2 != nil {
+					return fmt.Errorf("to: %w", err2)
+				}
+				t.To = &to
+				amount, err2 := uncategorized.StringifyNativeDenomination(&body.Amount)
+				if err2 != nil {
+					return fmt.Errorf("amount: %w", err2)
+				}
+				t.Amount = &amount
+				return nil
+			},
+			ConsensusAccountsDeposit: func(body *consensusaccounts.Deposit) error {
+				to, err2 := uncategorized.StringifySdkAddress(body.To)
+				if err2 != nil {
+					return fmt.Errorf("to: %w", err2)
+				}
+				t.To = &to
+				amount, err2 := uncategorized.StringifyNativeDenomination(&body.Amount)
+				if err2 != nil {
+					return fmt.Errorf("amount: %w", err2)
+				}
+				t.Amount = &amount
+				return nil
+			},
+			ConsensusAccountsWithdraw: func(body *consensusaccounts.Withdraw) error {
+				to, err2 := uncategorized.StringifySdkAddress(body.To)
+				if err2 != nil {
+					return fmt.Errorf("to: %w", err2)
+				}
+				// todo: is this right? we don't otherwise register this off-chain .To
```

Comment:
"here this is" is the right conclusion, I think :)
Since we're overloading the `To` field, let's just be more forthcoming in our API documentation, then. Currently, we just say "the meaning varies". I left a suggested edit that tries to break down the meaning; anything along those lines is good.

Reply:
comment replaced. continuing in the api spec conversations